File: second-year/second-semester/MPP/mpp-proiect-repository-raulbaciulescu/travelAgency2/persistance/UserRepository.cs

using model;
using model.Utils;
using persistance.Database;
using travelAgency2.Repository.Database;


namespace persistance
{
    public class UserRepository : IRepository<long, User>
    {
        private UserTable table;

        public UserRepository(TableFactory factory)
        {
            table = (UserTable) factory.GetTable(Db.Tables.USER);
         }
           
        public void Add(User entity)
        {
            table.Add(entity);
        }

        public void Delete(long id)
        {
            throw new NotImplementedException();
        }

        public User FindByID(long id)
        {
            return table.FindById(id);
        }

        public List<User> GetAll()
        {
            return table.GetAll();
        }

        public void Update(User entity, User entityNew)
        {
            throw new NotImplementedException();
        }

        public User FindUser(string username, string password)
        {
            return table.findUser(username, password);
        }
    }
}

File: second-year/first-semester/MAP/socialnetwork/src/main/java/com/example/socnet/repository/persistent/UserRepository.java
package com.example.socnet.repository.persistent;

import com.example.socnet.domain.data.MetaDB;
import com.example.socnet.domain.data.table.UserTable;
import com.example.socnet.domain.model.User;
import com.example.socnet.repository.memory.Repository;
import org.jetbrains.annotations.NotNull;

import java.sql.SQLException;
import java.util.ArrayList;
import java.util.Optional;

public class UserRepository extends Repository<Long, User> {

    private final UserTable table;

    public UserRepository() throws SQLException {
        this.table = (UserTable) MetaDB.getInstance().table("users");
        for (final User user : this.table.getAll()) {
            super.add(user);
        }
    }


    @Override
    public @NotNull Optional<User> add(@NotNull final User user) throws SQLException {
        final Optional<User> result = super.add(user);
        if (result.isPresent()) {
            this.table.insert(user);
        }
        return result;
    }

    @Override
    public @NotNull Optional<User> findByID(@NotNull final Long id) throws SQLException {
        return this.table.whereID(id);
    }

    @Override
    public @NotNull Optional<User> update(@NotNull final User user) throws SQLException {
        final Optional<User> result = super.update(user);
        if (result.isPresent()) {
            this.table.update(user);
        }
        return result;
    }

    @Override
    public @NotNull Optional<User> delete(@NotNull final Long id) throws SQLException {
        final Optional<User> result = super.delete(id);
        if (result.isPresent()) {
            this.table.delete(id);
        }
        return result;
    }

    @Override
    public @NotNull ArrayList<User> getAll() throws SQLException {
        return this.table.getAll();
    }
}
